.github/workflows/build.yml

```diff
@@ -3,17 +3,21 @@ name: Build Library
 on:
   push:
     branches: [main]
-    tags: ['v[0-9]+*']
   pull_request:
     branches: [main]
   workflow_dispatch:
+    inputs:
+      triggerRelease:
+        description: 'Release Build'
+        required: false
+        type: boolean
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
 env:
-  ORG_GRADLE_PROJECT_releaseBuild: ${{ github.event_name == 'push' && contains(github.ref, 'refs/tags/') }}
+  ORG_GRADLE_PROJECT_releaseBuild: ${{ github.event_name == 'workflow_dispatch' && inputs.triggerRelease && github.ref == 'refs/heads/main' }}
```

Comment:
this environment variable is used by the gradle build script to determine if this is a release or snapshot build